internal/nexodus/dump.go

```diff
@@ -33,54 +29,25 @@ func ShowDump(iface string) (string, error) {
 
 // DumpPeers dump wireguard peers
 func DumpPeers(iface string) ([]WgSessions, error) {
-	result, err := ShowDump(iface)
+	c, err := wgctrl.New()
 	if err != nil {
-		return nil, fmt.Errorf("error running wg show %s dump: %w", iface, err)
+		return nil, err
+	}
+	device, err := c.Device(iface)
+	if err != nil {
+		return nil, err
 	}
-	r := bufio.NewReader(strings.NewReader(result))
 	peers := make([]WgSessions, 0)
-	for {
-		line, _, err := r.ReadLine()
-		if err != nil {
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			return nil, fmt.Errorf("failed to read wg dump: %w", err)
-		}
-		column := strings.Split(string(line), "	")
-		if len(column) != 8 {
-			continue
-		}
-		publicKey := column[0]
-		psk := column[1]
-		endpoints := column[2]
-		allowedIPs := strings.Split(column[3], ",")
-		latestHandshake, err := strconv.Atoi(column[4])
-		if err != nil {
-			return nil, fmt.Errorf("latest handshake parse failed: %w", err)
-		}
-		latestHandshakeTime := time.Duration(0)
-		if latestHandshake != 0 {
-			latestHandshakeTime = time.Since(time.Unix(int64(latestHandshake), 0))
-		}
-		tx, err := strconv.Atoi(column[5])
-		if err != nil {
-			return nil, fmt.Errorf("transfer received parse failed: %w", err)
-		}
-		tr, err := strconv.Atoi(column[6])
-		if err != nil {
-			return nil, fmt.Errorf("transfer sent parse failed: %w", err)
-		}
+	for _, peer := range device.Peers {
 		peers = append(peers, WgSessions{
-			PublicKey:       publicKey,
-			PreSharedKey:    psk,
-			Endpoint:        endpoints,
-			AllowedIPs:      allowedIPs,
-			LatestHandshake: latestHandshakeTime.String(),
-			Tx:              tx,
-			Rx:              tr,
+			PublicKey:       peer.PublicKey.String(),
+			PreSharedKey:    peer.PresharedKey.String(),
+			Endpoint:        peer.Endpoint.String(),
```

Comment:
notice all the .String() conversions here... should we change the WgSessions to preserve the types more?

Reply:
Probably, there is a scenario where the peer.Endpoint() can be `none` but it just throws a debug error. Going to merge to get some burn in on this and we can followup in reconcileState with better error handling of types and how we want to act if at all for the (none) return scenario for Endpoint. Ty!